### DISCUSSION TITLE
Best Approach for Running Sequence Scripts

### DISCUSSION THREAD

**Discussion by Sheharyar570** [TPC-Council/HammerDB#755] [Q&A] [answered]
I'm running a sequence script, there are 2 ways we can run a sequence script:
1- Build schema once, and run the whole sequence (As shared in the code snippet [here](https://www.hammerdb.com/docs/ch09s09.html)).
```
#!/usr/bin/tclsh
puts "SETTING CONFIGURATION"
dbset db mssqls
diset tpcc mssqls_driver timed
diset tpcc mssqls_rampup 0
diset tpcc mssqls_duration 1
vuset logtotemp 1
loadscript
puts "SEQUENCE STARTED"
foreach z { 1 2 4 } {
puts "$z VU TEST"
vuset vu $z
vucreate
vurun
vudestroy
}
puts "TEST SEQUENCE COMPLETE"
```

2- Build schema for each run in the sequence.

Currently, I'm following the first one. But I've some questions?
1. How does it impact the overall run, our tables size increases after each run, wouldn't that impact our next benchmark run?
2. Would it be okay, if we followed the 2nd scenario (this would increase overall run time).

I'd be grateful, if you can answer these questions.

**Comment by sm-shaw** [comment, marked as answer]
Hi, the TPC-C workload is designed for growth, so the short answer is no you don't have to rebuild the schema before each run. I would recommend for MySQL/MariaDB restarting the database just once after the build, otherwise let it run. Here is an example I've used recently for MariaDB (this allows a purge between runs that keeps the results consistent).
**Build**
```
#!/bin/tclsh
# maintainer: Pooja Jain

puts "SETTING CONFIGURATION"
dbset db maria
dbset bm TPC-C

diset connection maria_host localhost
diset connection maria_port 3306
diset connection maria_socket /tmp/mariadb.sock

set vu 100
set warehouse 1000
diset tpcc maria_count_ware $warehouse
diset tpcc maria_num_vu $vu
diset tpcc maria_user root
diset tpcc maria_pass maria
diset tpcc maria_dbase tpcc
diset tpcc maria_storage_engine innodb
if { $warehouse >= 200 } {
diset tpcc maria_partition true
        } else {
diset tpcc maria_partition false
        }
puts "SCHEMA BUILD STARTED"
buildschema
puts "SCHEMA BUILD COMPLETED"
```
**Run** (you could of course use `incr 4` to increase the vu count by 4 each time if preferred)
```
#!/bin/tclsh
# maintainer: Pooja Jain

set tmpdir $::env(TMP)
puts "SETTING CONFIGURATION"
dbset db maria
dbset bm TPC-C

jobs profileid 5
giset commandline keepalive_margin 1200
giset timeprofile xt_gather_timeout 1200

diset connection maria_host 127.0.0.1
diset connection maria_port 3306
diset connection maria_socket /tmp/mariadb.sock
#
diset tpcc maria_user root
diset tpcc maria_pass maria
diset tpcc maria_dbase tpcc
diset tpcc maria_driver timed
diset tpcc maria_rampup 2
diset tpcc maria_duration 5
diset tpcc maria_allwarehouse false
diset tpcc maria_timeprofile false
diset tpcc maria_no_stored_procs false
diset tpcc maria_allwarehouse false
diset tpcc maria_timeprofile false
diset tpcc maria_purge true
puts "TEST STARTED"
foreach z { 1 2 4 8 12 16 20 24 28 32 36 40 44 48 52 56 60 64 68 72 76 80 84 88 92 96 100 104 108 112 116 120 } {
loadscript
vuset vu $z
vuset logtotemp 1
vucreate
tcstart
tcstatus
vurun
tcstop
vudestroy
}
puts "TEST COMPLETE"
```
and this is the resulting plot of the NOPM values

<img width="500" alt="perfprofile" src="https://github.com/user-attachments/assets/cda59d85-6837-42f7-ae33-27df9de8222b">
